lib/phoenix/router/scope.ex

```diff
@@ -27,6 +27,10 @@ defmodule Phoenix.Router.Scope do
     as      = Keyword.get(opts, :as, Phoenix.Naming.resource_name(plug, "Controller"))
     alias?  = Keyword.get(opts, :alias, true)
 
+    if to_string(as) == "static"  do
+      raise ArgumentError, "`static` is a reserved prefix. Supply a different prefix to the route using the `:as` option."
```

Comment:
Lets also show the controller/plug name they tried to use here. It will avoid confusion if folks aren't familiar with the named helper generation and they only passed `AppWeb.StaticController`.